lib/fetching/bot-master.js

```diff
@@ -63,31 +63,42 @@ BotMaster.prototype.stop = function() {
 BotMaster.prototype.kill = function(bot) {
   if (bot) {
     // Stop bot to ensure unbinding of listeners
-    this.disableBot(bot);
+    this.disableBot(bot, function() {});
     // Remove bot instance from list of bots
     var index = this.bots.indexOf(bot);
     if (index > -1) this.bots.splice(index, 1);
   } else {
     // Kill all bots
     for (var i in this.bots) {
-      this.disableBot(this.bots[i]);
+      this.disableBot(this.bots[i], function() {});
       this.bots[i] = null;
       delete this.bots[i];
     }
     this.bots = [];
   }
 };
 
+// This is called when a new bot is added, and when botmaster.start is called.
 // Enable bot from current process
-BotMaster.prototype.enableBot = function(bot) {
+BotMaster.prototype.enableBot = function(bot, callback) {
   // Bail if fetching is disabled
-  if (!this.enabled) return;
-  if (bot.source.enabled) bot.start();
+  if (!this.isFetching) return;
+  Source.findById(bot.source._id, function(err, source) {
+    if (err) return callback(err);
```

Comment:
Don't callback. You could, for example, just log a warning that bot could not be enabled because its source does not exist, and print the error message.